chain/state/statetree.go

```diff
@@ -187,6 +212,10 @@ func (st *StateTree) Revert() error {
 		return err
 	}
 
+	// TODO: should it clean actorcache? Probably yes
```

Comment:
oh yeah, this should probably clean the actor cache too. I wonder what problems we might encounter because of that

Reply:
maybe we only need to purge the cache of entries since the last snapshot though? so maybe it makes sense to have two caches?

Reply:
Yeah, it makes sense, you would have an old generation (that survived a snapshot) and new gen that can be rolled back.

The question is: is it more expensive to repopulate the cache that to maintain two of them.
Probably yes.

Another issue is that this could cause a fork (if this case was ever hit).

Reply:
There is no "old generation", because [`actorCache` is cleared during `Flush`](https://github.com/filecoin-project/lotus/blob/375a02c17119d20c158ad84a3f780b18b057d465/chain/state/statetree.go#L154) which is called by `Snapshot`

Reply:
Yes, that was a hypothetical development.